packages/ui/src/components/Select/select.tsx

```diff
@@ -9,28 +10,54 @@ import type { TypographyTextProps } from '@open-condo/ui/src'
 import { sendAnalyticsChangeEvent, extractChildrenContent } from '@open-condo/ui/src/components/_utils/analytics'
 
 import { SELECT_TYPES } from './constants'
+import { useItems } from './hooks/useItems'
+
+import { Either } from '../_utils/types'
 
 import type { SelectProps as DefaultSelectProps } from 'antd'
 
-const SELECT_CLASS_PREFIX = 'condo-select'
-const Option: typeof DefaultSelect.Option = DefaultSelect.Option
-const OptGroup: typeof DefaultSelect.OptGroup = DefaultSelect.OptGroup
+export const SELECT_CLASS_PREFIX = 'condo-select'
+export const Option: typeof DefaultSelect.Option = DefaultSelect.Option
+export const OptGroup: typeof DefaultSelect.OptGroup = DefaultSelect.OptGroup
 
-export type OptionType = React.PropsWithChildren<{
-    label: React.ReactNode
+export type OptionType = {
+    label: string
     value?: string | number | null
     disabled?: boolean
     textType?: TypographyTextProps['type']
     title?: string
+    key?: React.Key
+    hidden?: boolean
+}
+
+export type OptionsGroupType = {
+    label: string
+    options: Array<OptionType>
+    key?: React.Key
+}
+
+export type OptionsItem = Either<OptionType, OptionsGroupType> | null
+
+type OnChangeType = (
+    value: OptionType['value'] | Array<OptionType['value']>,
+    option: React.PropsWithChildren<OptionType> | Array<React.PropsWithChildren<OptionType>>
+) => void
+
+/**
+ * NOTE: Ant have bad interface for Select
+ * It may work with multiple values, but it is not specified in the interface.
+ */
+type CustomSelectProps<ValueType = SelectValueTypeBase> = Either<{
+    value?: ValueType | null
+}, {
+    mode: 'multiple'
+    value?: ValueType[] | null
```

Comment:
Af far as I remember we prefer to use `Array<Type>` instead of `Type[]`

Reply:
fixed